### DISCUSSION TITLE
Replacement for Sensio\Bundle\FrameworkExtraBundle\Configuration\ConfigurationInterface

### DISCUSSION THREAD

**Discussion by gnumoksha** [symfony/symfony#49530] [Q&A] [answered]
Is there a replacement for Sensio\Bundle\FrameworkExtraBundle\Configuration\ConfigurationInterface in order to create a class attribute which will be read by the framework?

**Comment by MatTheCat** [comment]
[Attributes](https://www.php.net/manual/fr/language.attributes.overview.php)?

**Reply by gnumoksha** [reply]
I'm looking for a replacement for `Sensio\Bundle\FrameworkExtraBundle\Configuration\ConfigurationInterface`, not for Sensio FrameworkExtraBundle itself. That interface enables reading controller attributes.

**Reply by MatTheCat** [reply]
Well you don’t need any interface to read classes attributes using native PHP. What do you want to do?

**Reply by gnumoksha** [reply]
That interface enables reading controller attributes without the need to do the reflection again.

**Comment by mbabker** [comment]
There isn't a direct replacement.  `Sensio\Bundle\FrameworkExtraBundle\Configuration\ConfigurationInterface` acted as a marker interface for annotations/attributes supported by that bundle and an event listener transformed that information into stuff usable by that bundle or its custom add-ons.  Depending on what you're trying to do, you'll need to add your own classes to replace that functionality.

**Reply by stof** [reply]
The `ControllerEvent` and `ControllerArgumentsEvent` events already provide a way to read the attributes that are set on the controller (taking care of optimizing reflection reads). And this code does not require a specific interface.

**Reply by gnumoksha** [reply]
Thank you, @stof! That worked.

I can't mark your answer as the accepted one, though, as it is a comment.

**Comment by gnumoksha** [comment, marked as answer]
@stof said:

> The `ControllerEvent` and `ControllerArgumentsEvent` events already provide a way to read the attributes that are set on the controller (taking care of optimizing reflection reads). And this code does not require a specific interface.
